src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapperUtil.java

```diff
@@ -193,7 +193,7 @@ private static int getEfConstruction(Settings indexSettings, Version indexVersio
         return Integer.parseInt(efConstruction);
     }
 
-    static KNNMethodContext createKNNMethodContextFromLegacy(Settings indexSettings, Version indexCreatedVersion) {
+    public static KNNMethodContext createKNNMethodContextFromLegacy(Settings indexSettings, Version indexCreatedVersion) {
```

Comment:
I though we are replacing MethodContext with IndexContext ?

Reply:
I see IndexContext is also having engine and SpaceType , Can we completely get rid of MethodContext Class,It's confusing?

Reply:
The idea was to ensure that there was a difference between a top level and a component. I think we could potentially get rid of it but might hold on for now.